instructor/common.py

```diff
@@ -200,34 +207,39 @@ def assignment_modify(request):
 def course_modify(request):
     assignment_id = request.session['assignment_id']
     assignment = get_object_or_404(models.Assignment, pk=assignment_id)
-    CourseFormSet = modelformset_factory(models.Course, extra=1, fields=['name', 'code'])
+    CourseFormSet = modelformset_factory(models.Course, extra=1, can_delete=True, fields=['name', 'code'])
     if request.method == 'POST':
-        if 'course_pk' in request.POST:
+        pk = request.POST['course_pk']
+        if not long(pk) == assignment.course.id:
             # change the course for this assignment
-            new_course = models.Course.objects.get(pk=request.POST['course_pk'])
+            new_course = get_object_or_404(models.Course, pk=int(pk))
             assignment.course = new_course
             assignment.save()
+
+        # change this course's name or code
+        formset = CourseFormSet(request.POST)
+        if formset.is_valid():
+
+            instances = formset.save(commit=False)
```

Comment:
I think maybe we could display the not owned courses in a non-modifiable way, so that the instructor knows from the begging that he cannot modify courses he doesn't own, and there won't be an option for that.